src/js/components/TextInput/TextInput.js

```diff
@@ -444,12 +466,35 @@ const TextInput = forwardRef(
 
     const textInputIcon = useSizedIcon(icon, rest.size, theme);
 
+    const CopyButton = (
+      <Tip dropProps={{ align: { bottom: 'top' } }} content={tipContent}>
+        <Button
+          onClick={copyOnClick}
+          plain
+          icon={<CopyIcon />}
+          margin={reverse ? { left: 'small' } : { right: 'small' }}
+          onBlur={copyOnBlur}
+          onMouseOut={copyOnBlur}
+          aria-label={`${format({
+            id: 'input.readOnlyCopy',
+            messages,
+          })} ${value}`}
+        />
+      </Tip>
+    );
+
     return (
-      <StyledTextInputContainer plain={plain}>
+      <StyledTextInputContainer
+        readOnlyCopy={readOnlyCopy}
+        plain={plain}
+        border={!plain}
+        round={theme.dateInput.container.round}
```

Comment:
It seems odd to have dateInput theme here

Reply:
Oops this was a mistake, I removed it